tracer_options.go

```diff
@@ -0,0 +1,170 @@
+package zipkintracer
+
+import (
+	"net"
+
+	"github.com/openzipkin/zipkin-go/model"
+
+	otobserver "github.com/opentracing-contrib/go-observer"
+	"github.com/openzipkin/zipkin-go"
+)
+
+// Endpoint holds the network context of a node in the service graph.
+type Endpoint struct {
+	ServiceName string
+	IPv4        net.IP
+	IPv6        net.IP
+	Port        uint16
+}
+
+func (e *Endpoint) toZipkin() *model.Endpoint {
+	return &model.Endpoint{
+		ServiceName: e.ServiceName,
+		IPv4:        e.IPv4,
+		IPv6:        e.IPv6,
+		Port:        e.Port,
+	}
+}
+
+// TracerOptions allows creating a customized Tracer.
+type TracerOptions struct {
+	localEndpoint *model.Endpoint
+
+	// shouldSample is a function which is called when creating a new Span and
+	// determines whether that Span is sampled. The randomized TraceID is supplied
+	// to allow deterministic sampling decisions to be made across different nodes.
+	sampler zipkin.Sampler
+
+	// unsampledNoop turns potentially expensive operations on unsampled
```

Comment:
why have this? it seems half of the sampled local thing...

Reply:
not sure I understand the question. it is here because of compatibility but also because it is supported in zipkin-go unless you refer to the actual description.

Reply:
actually why do we need to copy options from zipkin-go anyway? could you not have a funciton that takes a zipkin-go tracer (with options already applied) plus OT relevant options? It seems less maintenance.. or am I missing something?

Reply:
or by compatibility do you mean that this option was here before so that's why. Anyway, I'll leave you to it. I think you know I find this api surprising especially to someone using OT when they can just use zipkin-go directly if worried about perf. I would be surprised if layered instrumentation do smart things routinely, that's all.

Reply:
Right, now I understand the question :). So, this option was already here: https://github.com/openzipkin-contrib/zipkin-go-opentracing/blob/master/tracer.go#L36 I just made the translation into the counter part in zipkin-go.

I have removed many options that were here (most of them related to logs) and did not match zipkin-go but those easily having a counter part I kept.

BTW there is a method for wrapping an existing tracer: https://github.com/openzipkin-contrib/zipkin-go-opentracing/pull/123/files#diff-46db78603208f38452c0333bec12ab2cR57

I don't have strong feelings on keeping it or removing it but since we don't add any logic for this but to translate into zipkin option I'd rather keep it for retrocompatibility purposes.